src/h5web/metadata-viewer/utils.ts

```diff
@@ -1,7 +1,45 @@
+import {
+  HDF5BaseType,
+  HDF5Endianness,
+  HDF5TypeClass,
+} from '../providers/hdf5-models';
+
 export function renderShapeDims(dims: number[]): string {
   if (dims.length > 1) {
     return `${dims.join(' x ')} = ${dims.reduce((acc, value) => acc * value)}`;
   }
 
   return dims.toString();
 }
+
+export function renderEndianness(endianness: HDF5Endianness): string {
+  switch (endianness) {
+    case 'LE':
+      return 'Little-endian';
+
+    case 'BE':
+      return 'Big-endian';
+
+    case 'Native':
+      return 'Native';
+
+    case 'Not applicable':
+      return '';
+
+    default:
+      return '';
+  }
+}
+
+export function renderBaseType(type: HDF5BaseType): string {
+  if (type.class === HDF5TypeClass.String) {
+    const { length, charSet } = type;
+    return `${charSet} string (size:
+      ${Number.isInteger(length) ? length : 'variable'})`;
```

Comment:
I would keep the term `length` and display "variable length" rather than "length: variable". Perhaps you could also replace the parens with a simple colon: `UTF8 string, variable length`.

Reply:
What about `string, variable length` or `string, n characters` ?

Reply:
How about `n chars` so it's a bit more concise?